winpr/libwinpr/crypto/cipher.c

```diff
@@ -111,18 +126,19 @@ WINPR_RC4_CTX* winpr_RC4_New(const BYTE* key, size_t keylen)
 
 BOOL winpr_RC4_Update(WINPR_RC4_CTX* ctx, size_t length, const BYTE* input, BYTE* output)
 {
+	WINPR_ASSERT(ctx);
 #if defined(WITH_OPENSSL)
+	WINPR_ASSERT(ctx->u.ctx);
 	int outputLength;
 	if (length > INT_MAX)
 		return FALSE;
 
-	WINPR_ASSERT(ctx);
-	if (EVP_CipherUpdate((EVP_CIPHER_CTX*)ctx, output, &outputLength, input, (int)length) != 1)
-		return FALSE;
+	EVP_CipherUpdate(ctx->u.ctx, output, &outputLength, input, (int)length);
```

Comment:
I don't know much about the handling here, however, is there a reason for removing this check here?

Reply:
+1

Reply:
no, that was an accident.